lib/auth.tsx

```diff
@@ -0,0 +1,111 @@
+import React, {useState, useContext, createContext, useEffect} from 'react'
+import {
+  ApolloProvider,
+  ApolloClient,
+  InMemoryCache,
+  HttpLink,
+  gql,
+} from '@apollo/client'
+import useStorage from "./useStorage";
+
+
+// @ts-ignore
+const authContext:any = createContext()
+
+export function AuthProvider({ children}:any) {
+  const auth = useProvideAuth()
+
+  return (
+    <authContext.Provider value={auth}>
+      <ApolloProvider client={auth.createApolloClient()}>
+        {children}
+      </ApolloProvider>
+    </authContext.Provider>
+  )
+}
+
+export const useAuth:any = () => {
+  return useContext(authContext)
+}
+
+function useProvideAuth() {
+  const {getItem, setItem } = useStorage()
+  const [authToken, setAuthToken] = useState(null as string | null)
+  const [authId, setAuthId] = useState(null as string | null)
+  const storedToken =  getItem('token', 'local') !== ''? getItem('token', 'local') : null
+  useEffect(()=>setAuthToken(storedToken), [])
+  const storedId =  getItem('authId', 'local') !== ''? getItem('authId', 'local') : null
+  useEffect(()=>setAuthId(storedId), [])
+
+
+  const isSignedIn = () => {
+    if (authToken) {
+      return true
+    } else {
+      return false
+    }
+  }
+
+  const getAuthHeaders = () => {
+    if (!authToken) return null
+
+    return {
+      authorization: `Bearer ${authToken}`,
+    }
+  }
+
+  const createApolloClient = () => {
+    const link = new HttpLink({
+      uri: "https://reflow-demo.dyne.org/api/graphql",
```

Comment:
can you please put it as an .env variable... the endpoint (that will change for sure)